### PR TITLE
#607: hidden identifier work and framework for future match warning work

### PR DESCRIPTION
**Pull request by echicoine-icf**
This work encompasses the ongoing development of methodology for maintaining a list of exact matches, case insensitive matches, and lays ground work for future "sounds like" matches.

### CODE REVIEW

File: Src/java/cql-to-elm/src/test/resources/org/cqframework/cql/cql2elm/CMS146v2_Expected.json [hunk 1]
```diff
@@ -3,6 +3,36 @@
     "annotation" : [ {
       "translatorOptions" : "",
       "type" : "CqlToElmInfo"
+    }, {
```
Comment (echicoine-esac) [inline Src/java/cql-to-elm/src/test/resources/org/cqframework/cql/cql2elm/CMS146v2_Expected.json:6]: This file ended up being the perfect test for the hidden identifier work

**Comment by echicoine-esac** [comment]
> This PR now has conflicts. Can you resolve those?

Conflicts should be resolved currently.

**Comment by JPercival** [comment]
I do understand the need for the `hiddenIdentifier` work here, but I don't understand the use case for "sounds like" matches in a compile. Would you mind elaborating on that?

**Comment by JPercival** [comment]
Closed in favor of #1277

**Pull request closed without merging**